felix/iptables/table.go

```diff
@@ -677,7 +678,8 @@ func (t *Table) loadDataplaneState() {
 		if !t.ourChainsRegexp.MatchString(chainName) {
 			// Not one of our chains so it may be one that we're inserting rules into.
 			insertedRules := t.chainToInsertedRules[chainName]
-			if len(insertedRules) == 0 {
+			appendedRules := t.chainToAppendedRules[chainName]
+			if len(insertedRules) == 0 && len(appendedRules) == 0 {
 				// This chain shouldn't have any inserts, make sure that's the
```

Comment:
```suggestion
				// This chain shouldn't have any inserts or appends, make sure that's the
```

Reply:
IIUC, the problem is that we're treating our own appended rules as if they are unexpected, so we only want to hit this branch if we have neither inserts nor appends expected?

Reply:
Yes, exactly.  So we report them as unexpected, then we compare hashes below and find out that it's all correct anyway.